robolectric-shadows/shadows-core/src/main/java/org/robolectric/fakes/RoboAttributeSet.java

```diff
@@ -2,242 +2,74 @@
-    return resourceId == null ? defaultValue : resourceId;
-  }
-
-  @Override
-  public int getAttributeResourceValue(int resourceId, int defaultValue) {
-    String attrName = context.getResources().getResourceName(resourceId);
-    ResName resName = getAttrResName(null, attrName);
-    Attribute attr = findByName(resName);
-    if (attr == null) return defaultValue;
-    Integer extracted = ResName.getResourceId(resourceLoader.getResourceIndex(), attr.value, attr.contextPackageName);
-    return (extracted == null) ? defaultValue : extracted;
-  }
-
-  @Override
-  public int getAttributeIntValue(int index, int defaultValue) {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public int getAttributeUnsignedIntValue(int index, int defaultValue) {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public float getAttributeFloatValue(int index, float defaultValue) {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public String getIdAttribute() {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public String getClassAttribute() {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public int getIdAttributeResourceValue(int defaultValue) {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override public int getStyleAttribute() {
-    Attribute styleAttribute = find(attributes, new ResName("", "attr", "style"));
-    if (styleAttribute == null) {
-      // Per Android specifications, return 0 if there is no style.
-      return 0;
-    }
-    Integer i = ResName.getResourceId(resourceLoader.getResourceIndex(), styleAttribute.value, styleAttribute.contextPackageName);
-    return i != null ? i : 0;
-  }
-
-  private ResName getAttrResName(String namespace, String attrName) {
-    String packageName = Attribute.extractPackageName(namespace);
-    return new ResName(packageName, "attr", attrName);
-  }
-
-  private Attribute findByName(ResName resName) {
-    ResourceIndex resourceIndex = resourceLoader.getResourceIndex();
-    Integer resourceId = resourceIndex.getResourceId(resName);
-    // canonicalize the attr name if we can, otherwise don't...
-    // todo: this is awful; fix it.
-    if (resourceId == null) {
-      return find(attributes, resName);
-    } else {
-      return find(attributes, resourceId, resourceIndex);
-    }
-  }
-
-  private static Attribute find(List<Attribute> attributes, ResName resName) {
-    for (Attribute attribute : attributes) {
-      if (resName.equals(attribute.resName)) {
-        return attribute;
+    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+    factory.setNamespaceAware(true);
+    factory.setIgnoringComments(true);
+    factory.setIgnoringElementContentWhitespace(true);
+    Document document;
+    try {
+      DocumentBuilder documentBuilder = factory.newDocumentBuilder();
```

Comment:
This seems likely to be quite a bit slower than the old impl, but I think that doesn't matter cuz this is will only be used for narrow stuff like unit testing views, right?